pkg/rbd/nodeserver.go

```diff
@@ -543,6 +543,19 @@ func (ns *NodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
 	if err != nil {
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
+	// volumePath is targetPath for block PVC and stagingPath for filesystem.
+	// check the path is mountpoint or not, if it is
+	// mountpoint treat this as block PVC or else it is filesystem PVC
+	// TODO remove this once ceph-csi supports CSI v1.2.0 spec
+	notMnt, err := mount.IsNotMountPoint(ns.mounter, volumePath)
+	if err != nil {
+		if !os.IsNotExist(err) {
```

Comment:
I am unable to reason out why we treat `IsNotExist` error as not an error and proceed. What I get is that, this will return `notMnt` as true, and then the call at line 559 will error out as the parent path of the stash is absent. Should we be treating this error specially here?

Reply:
updated to handle IsNotExist